### ISSUE TITLE
Typescript error: Expected 0 arguments, but got 1.ts(2554) (parameter) url: string

### ISSUE THREAD

**Issue by kmvan** [charpeni/react-native-url-polyfill#369]
```ts
import { URL } from 'react-native-url-polyfill'

const urlObj = new URL(url) // Expected 0 arguments, but got 1.ts(2554) (parameter) url: string
```

Any idea?

**Comment by charpeni** [comment]
What's the type for `NativeURL` do you have?

https://github.com/charpeni/react-native-url-polyfill/blob/29942dda661fceb1a0307249bcacca177f5b79b9/index.d.ts#L2

Mine is from `@types/node/url.d.ts`.

**Comment by SnowLew** [comment]
Same error with me
![Captura de Tela 2021-11-11 às 20 50 09](https://user-images.githubusercontent.com/38509926/141389816-cb6d0423-d0f9-4229-b11b-c4dd42dda42a.png)
:

**Comment by jspizziri** [comment]
I've been using this pollyfill for quite some time in my project. I upgraded a few RN packages (not RN itself or anything to do with typings or this behavior) and all the sudden I started getting the same error (and many more related to the polyfill seeming to not exist anymore in my project).

**Comment by jspizziri** [comment]
I reverted my dependency upgrades and I can confirm that the polyfill is working again. So it has something to do with a conflict of dependencies. Here are the packages that I updated (they're all peerDependencies of my lib):

```
@notifee/react-native@^3.0.2 -> @^5.3.0
@react-native-community/netinfo@^6.0.2 -> ^9.0.0
@react-native-firebase/app@^12.9.3 -> ^14.11.0
@react-native-firebase/messaging@^12.9.3 -> ^14.11.0
react-native-bootsplash@^3.2.6 -> ^4.1.5
react-native-gesture-handler@^1.9.0 -> ^2.4.2
react-native-mmkv@^1.3.2 -> ^2.4.1
react-native-safe-area-context@^3.2.0 -> ^4.1.4
react-native-svg@^12.1.1 // newly installed
react-native-tab-view@^3.1.1 // newly installed
```

**Comment by raykle** [comment]
Hi @jspizziri, I'm facing the issue, and I finally resolve the issue by adding the code below at the first line of `index.d.ts` through `patch-package`. I tried this by referring to other packages.

```diff
+/// <reference types="node" />
 declare module 'react-native-url-polyfill' {
```

Happy to know the reason if somebody knows.

**Comment by 17Amir17** [comment]
Im having the same issue

**Comment by pasquale95** [comment]
Hi, I face the this issue when working with this lib. Additionally, such issue does not allow to retrieve the fields from the URL object. The solution from @raykle seems to fix the issue. 
@raykle can you make a PR with your fix so it gets merged?

PS: The solution from @ad-walker works with JS, but not with TS where I still face the issue.

**Comment by raykle** [comment]
@pasquale95 Yes I have submitted a PR.

**Comment by charpeni** [comment]
I'll close this issue as it was resolved by #444, thank you, @raykle!

It will soon be published on npm.
